server/syntax_gen.js

```diff
@@ -0,0 +1,86 @@
+// This file is part of the Soletta Project
+//
+// Copyright (C) 2016 Intel Corporation. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+module.exports = function () {
+    var fs = require('fs');
+    var path = require('path');
+    var nodes = require('./nodetypes/nodetypes.js');
+
+    var file_path = path.join(__dirname, '../client/libs/ace-builds/'+
+                                            'src-min/snippets/fbp.js');
+
+    this.generateNodetypes = function() {
+        try {
+            var content = nodes.getNodeTypesName();
+            var final = 'define(\"ace/snippets/fbp\",[\"require\"'+
+                        ',\"exports\",\"module\"],function(e,t,n)'+
+                        '{\"use strict\";t.snippetText=\"';
+
+            generateSnippets(content, final);
+        } catch (err) {
+            console.log(err);
+        }
+    };
+    
+    var generateSnippets = function(content, final) {
+        var strings = generateStrings(content);
+        var contlen = content.length;
+        for (var i = 0; i < contlen; i++) {
+            final = final+ 'trigger '+ strings.triggers.pop()+ 
+            '\\n'+ 'snippet '+ strings.names.pop()+ '\\n\\t'+
+             content.pop()+ '\\n';
```

Comment:
There are a lot of occurences in this patch that uses + together with another thing such as this line

Replace it from:
```
final = final+ 'trigger '+ strings.triggers.pop()+ 
            '\\n'+ 'snippet '+ strings.names.pop()+ '\\n\\t'+
            content.pop()+ '\\n';
```

To:

```
final = final + 'trigger  '+ strings.triggers.pop() + 
            '\\n' + 'snippet ' + strings.names.pop() + '\\n\\t' +
            content.pop() + '\\n';
```